torch/csrc/jit/ivalue.h

```diff
@@ -161,7 +161,7 @@ struct IValue {
   : tag(Tag::Tensor), retainable(t.defined())  {
     // note: the undefined tensor is not refcounted, so while it
     // is tagged as a tensor, retainable is set to false.
-    as_tensor_impl = t.at::detail::TensorBase::detach();
```

Comment:
I feel like I'm missing something.  Why are you mentioning `detach` in python?  I mentioned the semantics change because "t.TensorBase::detach() meant "return the raw pointer, relinquishing ownership on it." and now you aren't doing that.

Reply:
Oh, I think your confusion is about what `unsafeReleaseTensorImpl` does. This function is NOT a decref (actually, there is no way to decref the smart pointer class Tensor, and rightly so). It is, "relinquish control of the TensorImpl* pointer, and return it.) I'm using `release()` and `get()` nomenclature consistent with how `unique_ptr` names them.